src/main/java/com/mraof/minestuck/util/Dialogue.java

```diff
@@ -0,0 +1,496 @@
+		this.responses = responses;
+	}
+	
+	public ResourceLocation getPath()
+	{
+		return path;
+	}
+	
+	public String getMessage()
+	{
+		return message;
+	}
+	
+	public String getAnimation()
+	{
+		return animation;
+	}
+	
+	public ResourceLocation getGuiPath()
+	{
+		return guiPath;
+	}
+	
+	public List<Response> getResponses()
+	{
+		return responses;
+	}
+	
+	/**
+	 * A Response contains all possible Dialogues that can be reached from the present Dialogue.
+	 * It contains the message that represents the Response, any Conditions/Triggers, the location of the next Dialogue, and a boolean determining whether the Response should be visible when it fails to meet the Conditions
+	 */
+	public static class Response
+	{
+		private final String response;
+		private final List<Condition> conditions;
+		private final List<Trigger> triggers;
+		private final ResourceLocation nextDialoguePath;
+		private final boolean hideIfFailed;
+		
+		public Response(String response, List<Condition> conditions, List<Trigger> triggers, ResourceLocation nextDialoguePath, boolean hideIfFailed)
+		{
+			this.response = response;
+			this.conditions = conditions;
+			this.triggers = triggers;
+			this.nextDialoguePath = nextDialoguePath;
+			this.hideIfFailed = hideIfFailed;
+		}
+		
+		public String getResponse()
+		{
+			return response;
+		}
+		
+		public List<Condition> getConditions()
+		{
+			return conditions;
+		}
+		
+		public List<Trigger> getTriggers()
+		{
+			return triggers;
+		}
+		
+		public ResourceLocation getNextDialoguePath()
+		{
+			return nextDialoguePath;
+		}
+		
+		public boolean isHideIfFailed()
+		{
+			return hideIfFailed;
+		}
+		
+		public boolean failsWhileImportant(LivingEntity entity, Player player)
+		{
+			return !matchesAllConditions(entity, player) && isHideIfFailed();
+		}
+		
+		public boolean matchesAllConditions(LivingEntity entity, Player player)
+		{
+			for(Condition condition : conditions)
+			{
+				if(!condition.type.conditions.test(entity, player, Pair.of(condition.getContent(), condition.getContentExtra())))
+				{
+					return false;
+				}
+			}
+			
+			return true;
+		}
+	}
+	
+	/**
+	 * A Condition controls whether a piece of dialogue will show up
+	 */
+	public static class Condition
+	{
+		public enum Type
+		{
+			CONSORT_TYPE((entity, player, content) ->
+					entity instanceof ConsortEntity consortEntity && content.first.equals(consortEntity.getConsortType().getName())
+			),
+			HAS_ITEM((entity, player, content) ->
+			{
+				ItemStack stack = findPlayerItem(content.first, player, parseIntFromContentExtra(content.second, 1));
+				return stack != null;
+			});
+			
+			private final TriPredicate<LivingEntity, Player, Pair<String, String>> conditions;
+			
+			Type(TriPredicate<LivingEntity, Player, Pair<String, String>> conditions)
+			{
+				this.conditions = conditions;
+			}
+		}
+		
+		private final Type type;
+		private final String content;
+		@Nullable
+		private final String contentExtra;
```

Comment:
I'm not a great fan of keeping these in their unparsed form. I suggest some polymorphism here, with each condition type having a class with fields for their content parsed into the right types. Same goes for the triggers.